packages/battery/battery/android/src/main/java/io/flutter/plugins/battery/BatteryPlugin.java

```diff
@@ -119,6 +112,7 @@ public void onReceive(Context context, Intent intent) {
             events.success("full");
             break;
           case BatteryManager.BATTERY_STATUS_DISCHARGING:
+          case BatteryManager.BATTERY_STATUS_NOT_CHARGING:
```

Comment:
Should this actually be a new status string? These mean slightly different things right?

Reply:
BatteryManager.BATTERY_STATUS_NOT_CHARGING happen 99% in emulator because they don't have physical battery. And if the battery not charging so it's discharging. What do you think?

Reply:
It will be only for android. iOS don't have this status. And I will add unknown status string to iOS and android.

Reply:
Do you have documentation on this flag? What I'm able to find isn't entirely clear :)

Reply:
Me too . It's added from API 1 . The safest choice it's like you said I must add new status.